internal/index_read.go

```diff
@@ -103,14 +102,15 @@ func readBins(r io.Reader, typ string) ([]Bin, *ReferenceStats, error) {
 	return bins, stats, nil
 }
 
-func readChunks(r io.Reader, n int32, typ string, buf [16]byte) ([]bgzf.Chunk, error) {
+func readChunks(r io.Reader, n int32, typ string) ([]bgzf.Chunk, error) {
 	if n == 0 {
 		return nil, nil
 	}
 	chunks := make([]bgzf.Chunk, n)
+	buf := make([]byte, 16)
```

Comment:
If this is really hitting you, you should benchmark this approach against `var buf [16]byte` and passing `buf[:]` to the `io.ReadFull` call below. Post the results from benchstat.